ke/compiler.py

```diff
@@ -151,7 +194,7 @@ def compile_concat(concat, macros):
 
 
 def is_not_empty(node):
-    return node != EMPTY and node != EMPTY_CONCAT
+    return node != EMPTY and node != EMPTY_CONCAT and not (isinstance(node, asm.Multiple) and node.max == 0)
```

Comment:
Looking at this, I wonder if we should just add an `.is_empty()` to all types of nodes...